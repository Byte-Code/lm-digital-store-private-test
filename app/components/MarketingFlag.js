import React, { Component } from 'react';
import ImmutablePropTypes from 'react-immutable-proptypes';
import glamorous from 'glamorous';

import MarketingBadge from './MarketingBadge';
import { getPromotions, filterPromotions } from '../utils/marketingUtils';

const Wrapper = glamorous.div({
  display: 'flex',
  justifyContent: 'space-between',
  alignItems: 'flex-end',
  marginBottom: 10
});

export default class MarketingFlag extends Component {
  static propTypes = {
    marketingAttributes: ImmutablePropTypes.map.isRequired,
    loyaltyProgram: ImmutablePropTypes.map.isRequired
  };

  renderBadges() {
    const { marketingAttributes, loyaltyProgram } = this.props;
    const promotions = getPromotions(marketingAttributes, loyaltyProgram);
    const filteredPromotions = filterPromotions(promotions);

    return filteredPromotions.map(p => <MarketingBadge promotion={p} key={p.get('code')} />);
  }

  render() {
    return (
      <Wrapper>
        {this.renderBadges()}
      </Wrapper>
    );
  }
}
