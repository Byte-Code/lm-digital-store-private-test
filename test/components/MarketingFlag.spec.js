import React from 'react';
import { fromJS, Map } from 'immutable';
import { mount } from 'enzyme';

import MarketingFlag from '../../app/components/MarketingFlag';

describe('MarketingFlag', () => {
  it('should render properly when promotions are active', () => {
    const marketingAttributes = fromJS({
      specialBadgeCode: 'PREZZO_STOCK',
      specialBadgeCodePeriodEndDate: '2020-10-14T00:00:00.000Z'
    });
    const loyaltyProgram = fromJS({
      type: 'DISCOUNT',
      value: 10,
      endDate: '2020-10-14T00:00:00.000Z'
    });
    const result = mount(
      <MarketingFlag marketingAttributes={marketingAttributes} loyaltyProgram={loyaltyProgram} />
    );
    expect(result).toMatchSnapshot();
  });

  it('should render an empty div when no promotions are active', () => {
    const marketingAttributes = Map();
    const loyaltyProgram = Map();
    const result = mount(
      <MarketingFlag marketingAttributes={marketingAttributes} loyaltyProgram={loyaltyProgram} />
    );
    expect(result).toMatchSnapshot();
  });
});
