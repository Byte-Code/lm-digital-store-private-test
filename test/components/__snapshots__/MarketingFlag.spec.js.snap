// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarketingFlag should render an empty div when no promotions are active 1`] = `
<MarketingFlag
  loyaltyProgram={Object {}}
  marketingAttributes={Object {}}
>
  <glamorous(div)>
    <div
      className="css-c7um26"
    />
  </glamorous(div)>
</MarketingFlag>
`;

exports[`MarketingFlag should render properly when promotions are active 1`] = `
<MarketingFlag
  loyaltyProgram={
    Object {
      "type": "DISCOUNT",
      "value": 10,
    }
  }
  marketingAttributes={
    Object {
      "specialBadgeCode": "PREZZO_STOCK",
    }
  }
>
  <glamorous(div)>
    <div
      className="css-c7um26"
    >
      <MarketingBadge
        promotion={
          Object {
            "code": "IDEAPIU",
            "value": 10,
          }
        }
      >
        <Component
          src="test-file-stub"
          value={10}
        >
          <glamorous(div)>
            <div
              className="css-13azwyo"
            >
              <glamorous(img)
                height="50px"
                mBottom="18px"
                src="test-file-stub"
                width="100px"
              >
                <img
                  className="css-1mc4zot"
                  src="test-file-stub"
                />
              </glamorous(img)>
              <glamorous(p)>
                <p
                  className="css-178lsms"
                >
                  -10% IDEAPIÃš
                </p>
              </glamorous(p)>
            </div>
          </glamorous(div)>
        </Component>
      </MarketingBadge>
    </div>
  </glamorous(div)>
</MarketingFlag>
`;
